autogen/agentchat/conversable_agent.py

```diff
@@ -175,7 +160,39 @@ def __init__(
         self.reply_at_receive = defaultdict(bool)
         self.register_reply([Agent, None], ConversableAgent.generate_oai_reply)
         self.register_reply([Agent, None], ConversableAgent.a_generate_oai_reply, ignore_async_in_sync_chat=True)
-        self.register_reply([Agent, None], ConversableAgent.generate_code_execution_reply)
+
+        # Setting up code execution.
+        # Do not register code execution reply if code execution is disabled.
+        if code_execution_config is not False:
+            # If code_execution_config is None, set it to an empty dict.
+            if code_execution_config is None:
+                warnings.warn(
```

Comment:
Should we use DeprecationWarning?

```python
warnings.warn("message goes here", DeprecationWarning)
```

Reply:
Planning to add deprecation warning in a future PR when containerized code executors are ready.